crates/ruff_linter/src/rules/refurb/rules/verbose_decimal_constructor.rs

```diff
@@ -116,12 +116,9 @@ pub(crate) fn verbose_decimal_constructor(checker: &mut Checker, call: &ast::Exp
             func, arguments, ..
         }) => {
             // Must be a call to the `float` builtin.
-            let Some(func_name) = func.as_name_expr() else {
+            if !checker.semantic().match_builtin_expr(func, "float") {
```

Comment:
I think this change negatively impacts performance. The idea was to have all cheap tests early. Calling `resolve_symbol` isn't cheap. It might be worth adding a "fast" path in the slow path of `match_builtin_expr` that exits early if `expr` is a name but doesn't match the builtin name (except if you want to support `from builtins import float as f`)


You may want to move this check further down. The same applies for other places where we now perform this check as the very first thing in the function.

Reply:
> except if you want to support `from builtins import float as f`

I do, and we already support that in several of our rules. I tried the fast path you're suggesting here in the initial version of this PR, and it caused a regression in the ecosystem check: https://github.com/astral-sh/ruff/pull/10919#issuecomment-2053725755

Also Charlie told me off for not getting Python's semantics correct :-)

Reply:
> You may want to move this check further down. The same applies for other places where we now perform this check as the very first thing in the function.

that makes sense, I'll look at moving these checks down